Network/Socket/Options.hsc

```diff
@@ -39,10 +43,10 @@ import Network.Socket.Types
 --
 -- The existence of a constructor does not imply that the relevant option
 -- is supported on your system: see 'isSupportedSocketOption'
-data SocketOption = SockOpt {
-    sockOptLevel :: !CInt
-  , sockOptName  :: !CInt
-  } deriving (Eq, Show)
+data SocketOption = SockOpt
+    !CInt -- ^ Option Level
```

Comment:
```suggestion
    !CInt -- Option Level
```

Reply:
@vdukhovni Why do you suggest this change? haddock creates doc if `-- ^` is used.

Reply:
> @vdukhovni Why do you suggest this change? haddock creates doc if `-- ^` is used.

The reason is that with the haddock comment `doctests` for the pull request fail for GHC 8.0, 8.2 and 8.4.
Since the data structure is private anyway, the haddock comments are not essential IMHO...